src/services/intelligence/preference_storage.py

```
@@ -0,0 +1,545 @@
+            created=created,
+            updated=updated,
+            confidence=float(row["confidence"]),
+            frequency=int(row["frequency"]),
+            last_used=last_used,
+            source=row.get("source", "user_correction"),
+        ),
+        context=context,
+    )
+
+
+def _decode_pref_value(stored: str) -> Any:
```

Ruff (0.15.11)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `_decode_pref_value`

(ANN401)